### PR TITLE
APR sorting, TVL filter, incentivised category

### LINKED ISSUE #334: filter for pools with rewards only
automatic incentivize filter

also check ordering

### LINKED ISSUE #338: order by APR broken
seems ordering by APR does not take ranges into account

### LINKED ISSUE #343: filter for pools with minimum TVL
see [here](https://github.com/balancer/frontend-v3/issues/462)

supply a numeric value for minimum TVL to the query `poolGetPools` which only returns pools with a TVL equal to or higher than the supplied value

### LINKED ISSUE #354: Add token APR
We need to be able to show individual APRs to allow clients to interpret the data on their own. That means adding / changing the "items" array into individual APRs with labels, which will make it easy to do any logic on.

I'm thinking something to standardise the response object as well, so clients don't have to handle typed fields.

```
aprs: [{
  name: "BAL staking",
  label: gauge,
  apr: XX (number in %, eg: 0.1 = 10%)
}]
```

* we get rid of range / total and just use apr as a value
* we store each individual APR, eg: individual tokens, so we know where each APR is what, where it's coming from and it's easy for clients to interpret.
* we keep it backwards compatible, and prepare a migration, so beets don't break

### LINKED ISSUE #362: remove range from the APRs
Use cases for ranges:

* BAL min / max – we can store just min, because max = min * 2.5
* beets – ?
* voting APR – we can store max, because the range is 0 - max

Removing will simplify the schema where expanding on types won't be necessary:

From:
```
{
  apr {
    ... on GqlPoolAprTotal {
      total
    }
    ... on GqlPoolAprRange {
      min
    }
}
```

to:
```
{
  apr {
    value
  }
}
```

### PR DESCRIPTION
**Pull request by gmbronco**
closes #343 
closes #338 
closes #334

### CODE REVIEW

File: modules/actions/pool/sync-incentivized-category.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { Chain } from '@prisma/client';
+import { prisma } from '../../../prisma/prisma-client';
+
+export const syncIncentivizedCategory = async (chain: Chain) => {
+    const poolsWithBalReward = await prisma.prismaPoolAprItem.findMany({
+        select: { poolId: true },
+        where: {
+            type: 'NATIVE_REWARD',
```
Comment (franzns) [inline modules/actions/pool/sync-incentivized-category.ts]: should also be on the incentivized list if it has thirdParty rewards

File: modules/actions/pool/sync-incentivized-category.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { Chain } from '@prisma/client';
+import { prisma } from '../../../prisma/prisma-client';
+
+export const syncIncentivizedCategory = async (chain: Chain) => {
+    const poolsWithBalReward = await prisma.prismaPoolAprItem.findMany({
```
Comment (franzns) [inline modules/actions/pool/sync-incentivized-category.ts]: its poolsWithReward not specific to bal

File: modules/pool/pool.gql [hunk 3]
```diff
@@ -1077,6 +1095,31 @@ type GqlBalancePoolAprItem {
     subItems: [GqlBalancePoolAprSubItem!]
 }
 
+enum GqlAprType {
+    IB_YIELD # Interest bearing yield? I have reservations about that name.
```
Comment (franzns) [inline modules/pool/pool.gql]: call it token_yield

File: modules/pool/pool.resolvers.ts [hunk 4]
```diff
@@ -5,6 +5,31 @@ import { prisma } from '../../prisma/prisma-client';
 import { networkContext } from '../network/network-context.service';
 import { headerChain } from '../context/header-chain';
 import { EventsQueryController } from '../controllers/pool-events-query-controller';
+import { FieldNode, SelectionNode } from 'graphql';
+
+const extractPaths = (node: FieldNode) => {
```
Comment (franzns) [inline modules/pool/pool.resolvers.ts]: this isnt used, is it?

File: modules/pool/pool.gql [hunk 5]
```diff
@@ -1077,6 +1095,31 @@ type GqlBalancePoolAprItem {
...
+    SWAP_FEE
+    PHANTOM_STABLE_BOOSTED
+    VOTING
+    LINEAR_BOOSTED
+    NATIVE_REWARD
+    THIRD_PARTY_REWARD
+}
+
+type GqlPoolAprV2 {
+    total: Float!
```
Comment (franzns) [inline modules/pool/pool.gql]: that doesnt work for a range

File: modules/pool/lib/pool-gql-loader.service.ts [hunk 6]
```diff
@@ -84,10 +87,11 @@ export class PoolGqlLoaderService {
         );
     }
 
-    public async getPools(args: QueryPoolGetPoolsArgs): Promise<GqlPoolMinimal[]> {
+    public async getPools(args: QueryPoolGetPoolsArgs, paths: string[] = []): Promise<GqlPoolMinimal[]> {
         // only include wallet and staked balances if the query requests it
         // this makes sure that we don't load ALL user balances when we don't filter on userAddress
         // need to support ordering and paging by userbalanceUsd. Need to take care of that here, as the DB does not (and should not) store the usd balance
+        // TODO: let's find a way to do that in the DB
```
Comment (franzns) [inline modules/pool/lib/pool-gql-loader.service.ts]: I dont think we can for the simple fact that we dont store userBalances in USD (and for good reason)

I think the gql-loader service would need a general refactor.

**Comment by franzns** [comment]
Remove the APR changes, we'll need to think about that more. Rest is ok to stay, also nice you found a way to extract the requested fields.

**Comment by gmbronco** [comment]
sounds good, I removed the APR response changes and added all reward tokens to the `incentivised` category.

**Pull request merged**